### PR TITLE
[5x backport]: add support for ipv6

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Add support for ipv6.
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by kalensk** [comment]
Need to generate the 5X pipeline file (`5X_STABLE-generated.yml`) and commit it.

**Comment by kalensk** [comment]
Just a reminder to re-fly the production pipeline after merging to pickup the new gpcheckperf tests.

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@
                                 'filerep_end_to_end_incr_primary'] %}
 {% set MM_1_suites = ['gprecoverseg',
                       'gpcheckcat',
+                      'gpcheckperf',
```
Comment (kalensk) [inline concourse/pipelines/templates/gpdb-tpl.yml:21]: Unfortunately, for 5X I believe we need to "manually" add the CI job (unlike for 6X and 7X)...

File: gpAux/platform/gpnetbench/gpnetbenchClient.c [hunk 2]
```diff
@@ -125,38 +123,58 @@ int main(int argc, char** argv)
 	{ 
 		fprintf(stdout, "socket call failed\n");
 		return 1;
-	}   
+	}
```
Comment (dkrieger-pvtl) [inline gpAux/platform/gpnetbench/gpnetbenchClient.c:127]: https://github.com/greenplum-db/gpdb/pull/11350/files#r562979054

remove the original socket creation.

File: gpAux/platform/gpnetbench/gpnetbenchServer.c [hunk 3]
```diff
@@ -46,83 +37,118 @@ int main(int argc, char** argv)
...
+	struct addrinfo *addrs;
+	int s, clientFd, clientPid;
+	int one = 1;
+	int fd = -1;
+	int pid = -1;
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = protocol;	/* Allow IPv4 or IPv6 */
+	hints.ai_socktype = SOCK_STREAM;	/* Two-way, out of band connection */
+	hints.ai_flags = AI_PASSIVE;	/* For wildcard IP address */
```
Comment (dkrieger-pvtl) [inline gpAux/platform/gpnetbench/gpnetbenchServer.c:71]: As in [7X](https://github.com/greenplum-db/gpdb/pull/11350#discussion_r560601073) I think AI_PASSIVE should be removed here.

File: gpAux/platform/gpnetbench/gpnetbenchServer.c [hunk 4]
```diff
@@ -46,83 +37,118 @@ int main(int argc, char** argv)
...
+				{
+					fprintf(stderr, "failed to fork process");
+					exit(1);
+				}
+			}
+			else
+				close(fd);
+
+			addrs = addrs->ai_next;
+		}
```
Comment (dkrieger-pvtl) [inline gpAux/platform/gpnetbench/gpnetbenchServer.c:123]: The L81/L123 braces are an unnecessary additional code block.  Please remove them.

**Review by dkrieger-pvtl — COMMENTED** [review]
See two comments.

File: gpMgmt/test/behave/mgmt_utils/gpcheckperf.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    When  the user runs "gpcheckperf -h mdw -h sdw1 -d /data/gpdata/ -r ds"
+    Then  gpcheckperf should return a return code of 0
+    And   gpcheckperf should print "disk write tot bytes" to stdout
+
+  @concourse_cluster
+  Scenario: gpcheckperf runs runs sequential network test
+    Given the database is running
+    When  the user runs "gpcheckperf -h mdw -h sdw1 -d /data/gpdata/ -r n"
+    Then  gpcheckperf should return a return code of 0
+    And   gpcheckperf should print "avg = " to stdout
```
Comment (dkrieger-pvtl) [inline gpMgmt/test/behave/mgmt_utils/gpcheckperf.feature:16]: super nit: please add a newline here.

File: gpAux/platform/gpnetbench/gpnetbenchClient.c (hunk 2 above)
Comment (dkrieger-pvtl) [reply gpAux/platform/gpnetbench/gpnetbenchClient.c]: Done.

**Review by dkrieger-pvtl — APPROVED** [review]
LGTM.

**Pull request merged**